opencog/distvalue/DistributionalValue.cc

```diff
@@ -0,0 +1,287 @@
+/*
+ * opencog/truthvalue/DistributionalValue.cc
+ *
+ * Copyright (C) 2018 SingularityNet
+ * All Rights Reserved
+ *
+ * Written by Roman Treutlein
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <float.h>
+#include <math.h>
+#include <stdio.h>
+#include <tuple>
+#include <iomanip>
+
+#include <opencog/distvalue/DistributionalValue.h>
+#include <opencog/atomspace/AtomSpace.h>
+#include <opencog/types/atom_types.h>
+
+#include <boost/range/combine.hpp>
+
+using namespace opencog;
+
+count_t DistributionalValue::DEFAULT_K = 800.0;
+
+DistributionalValue::DistributionalValue()
+	: Value(DISTRIBUTIONAL_VALUE) , _value(15,1) {}
+
+DistributionalValue::DistributionalValue(size_t size,size_t dims)
+	: Value(DISTRIBUTIONAL_VALUE) , _value(size,dims) {}
+
+DistributionalValue::DistributionalValue(const CTHist<double> &hist)
+	: Value(DISTRIBUTIONAL_VALUE) , _value(hist) {}
+
+//Create a DV from the Parameters of a SimpleTV
+//Not recommended as it results in a DV with only 1 singleton set
+//which in some calculations is unusalbe as the chance of overlaps in the
+//Sets/Bins is small
+DistributionalValue::DistributionalValue(double mode,double conf,bool fuzzy)
+	: Value(DISTRIBUTIONAL_VALUE) , _value(fuzzy ? 1 : 2 ,1)
+{
+	confidence_t cf = std::min(conf, 0.9999998);
+	double count = (DEFAULT_K * cf / (1.0 - cf));
+	//DV Bin with count 0 is undefined
+	count = std::max(count, 0.0000002);
+
+	if (fuzzy)
+	{
+		_value.insert(DVec{mode},count);
```

Comment:
We follow the convention of adding a whitespace after a comma, as mentioned in https://wiki.opencog.org/w/Development_standards#Formatting